gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java

```diff
@@ -234,14 +235,16 @@ public StorageObject call() {
           }
         }, serviceOptions.retryParams(), EXCEPTION_HANDLER);
     String cursor = result.x();
-    return new BaseListResult<>(new BucketPageFetcher(serviceOptions, cursor, optionsMap), cursor,
-        Iterables.transform(result.y(),
+    Iterable<BucketInfo> buckets =
+        result.y() == null ? ImmutableList.<BucketInfo>of() : Iterables.transform(result.y(),
```

Comment:
Yes and yes. I am writing tests for the whole `StorageImpl` class but I have not finished yet, I did not want to push a WIP with this PR.